boa/src/builtins/regexp/mod.rs

```diff
@@ -188,114 +185,210 @@ impl RegExp {
     /// The amount of arguments this function object takes.
     pub(crate) const LENGTH: usize = 2;
 
-    /// Create a new `RegExp`
+    /// `22.2.3.1 RegExp ( pattern, flags )`
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-regexp-pattern-flags
     pub(crate) fn constructor(
         new_target: &Value,
         args: &[Value],
-        ctx: &mut Context,
+        context: &mut Context,
     ) -> Result<Value> {
-        let prototype = new_target
-            .as_object()
-            .and_then(|obj| {
-                obj.__get__(&PROTOTYPE.into(), obj.clone().into(), ctx)
-                    .map(|o| o.as_object())
-                    .transpose()
-            })
-            .transpose()?
-            .unwrap_or_else(|| ctx.standard_objects().regexp_object().prototype());
-        let this = Value::new_object(ctx);
-
-        this.as_object()
-            .expect("this should be an object")
-            .set_prototype_instance(prototype.into());
-        let arg = args.get(0).ok_or_else(Value::undefined)?;
-
-        let (regex_body, mut regex_flags) = match arg {
-            Value::Undefined => (
-                String::new().into_boxed_str(),
-                String::new().into_boxed_str(),
-            ),
-            Value::Object(ref obj) => {
-                let obj = obj.borrow();
-                if let Some(regex) = obj.as_regexp() {
-                    // first argument is another `RegExp` object, so copy its pattern and flags
-                    (regex.original_source.clone(), regex.original_flags.clone())
-                } else {
-                    (
-                        arg.to_string(ctx)?.to_string().into_boxed_str(),
-                        String::new().into_boxed_str(),
-                    )
+        let pattern = args.get(0).cloned().unwrap_or_else(Value::undefined);
+        let flags = args.get(1).cloned().unwrap_or_else(Value::undefined);
+
+        // 1. Let patternIsRegExp be ? IsRegExp(pattern).
+        let pattern_is_regexp = if let Value::Object(obj) = &pattern {
+            let obj = obj.borrow();
+            obj.is_regexp()
+        } else {
+            false
+        };
```

Comment:
```suggestion
        let pattern_is_regexp = if let Value::Object(obj) = &pattern {
            let obj = obj.borrow();
            obj.is_regexp()
        	Some(obj.clone())
        } else {
            None
        };
```

So we can avoid calling `.get_field`, we can let it be a `Option<GcObject>`